File: 26_Remove Duplicates From Sorted Array.java
// Time complexity: O(n)
// Space complexity: O(1)

class Solution {
    public int removeDuplicates(int[] nums) {
        // point to the last non-repeated digit
        int slow = 0;
        // iterate through the rest array
        for (int fast = 1; fast < nums.length; fast++) {
            if (nums[slow] != nums[fast]) {
                slow ++;
                nums[slow] = nums[fast];
            }
        }
        return slow + 1;
    }
}

File: 26_Remove Duplicates From Sorted Array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # 2-ptr w/ in-place swap
        l = 0
        for r in range(len(nums)):
            if nums[r] != nums[l]:
                l += 1
                nums[l] = nums[r]
        return l + 1